static/js/components/CreateCommentForm_test.js

```diff
@@ -49,124 +56,221 @@ describe("CreateCommentForm", () => {
+        ],
+        () => {
+          wrapper.find("form").simulate("submit")
+        }
+      )
+      assert.deepEqual(state.forms[replyToPostKey(post)], emptyFormState)
+      assert.equal(
+        state.posts.data.get(post.id).num_comments,
+        post.num_comments + 1
+      )
+      assert.isOk(
+        helper.createCommentStub.calledWith(
+          expectedKeys.post_id,
+          expectedKeys.text,
+          expectedKeys.comment_id || undefined
+        )
+      )
+    })
+  })
+
+  describe("ReplyToCommentForm", () => {
+    let expectedKeys, wrapper
+    beforeEach(() => {
+      (wrapper = renderCommentForm(comment)), (expectedKeys = commentKeys)
+    })
+    it("should render a reply button", () => {
+      assert.equal(wrapper.find("a").text(), "Reply")
+    })
+
+    it("should show an empty form when reply is clicked", () => {
+      return helper
+        .listenForActions([forms.FORM_BEGIN_EDIT], () => {
+          wrapper.find("a").simulate("click")
+        })
+        .then(() => {
+          const [textarea] = wrapper.find("textarea[name='text']")
+          assert.equal(textarea.value, "")
+        })
+    })
+
+    it("should trigger an update in state when text is input", () => {
+      return helper
+        .listenForActions([forms.FORM_BEGIN_EDIT], () => {
+          wrapper.find("a").simulate("click")
+        })
+        .then(() =>
+          helper
+            .listenForActions([forms.FORM_UPDATE], () => {
+              wrapper.find("textarea[name='text']").simulate("change", {
+                target: {
+                  name:  "text",
+                  value: expectedKeys.text
+                }
               })
-              .then(state => {
-                assert.deepEqual(state.forms, {})
-                assert.isNotOk(wrapper.find("textarea[name='text']").exists())
+            })
+            .then(state => {
+              assert.equal(R.keys(state.forms).length, 1)
+              assert.deepInclude(state.forms[R.keys(state.forms)[0]], {
+                value:  expectedKeys,
+                errors: {}
               })
-          )
-      })
+            })
+        )
+    })
 
-      it("should submit the form", () => {
-        const { requestType, successType } = actions.comments.post
-        helper.createCommentStub.returns(Promise.resolve(makeComment(post)))
-        return helper
-          .listenForActions([SET_POST_DATA, forms.FORM_BEGIN_EDIT], () => {
-            helper.store.dispatch(setPostData(post))
-            wrapper.find("a").simulate("click")
-          })
-          .then(() =>
-            helper
-              .listenForActions([forms.FORM_UPDATE], () => {
-                wrapper.find("textarea[name='text']").simulate("change", {
-                  target: {
-                    name:  "text",
-                    value: expectedKeys.text
-                  }
-                })
+    it("should cancel and hide the form", () => {
+      return helper
```

Comment:
Should this and the following tests get refactored to the `async`/`await` styles while you're touching them here?

Reply:
probably not a bad idea, I'll do that now

Reply:
I really prefer the `async / await` sugar, just avoiding all those `.then` nested callbacks makes the code so much more readable IMO

Reply:
agreed